AppCache removal: remove applicationCache property

annevk opened the pull request:
#### 0f1d0a1df59cae58c89e3644433ccedc61659974
<pre>
AppCache removal: remove applicationCache property
<a href="https://bugs.webkit.org/show_bug.cgi?id=275312">https://bugs.webkit.org/show_bug.cgi?id=275312</a>

Reviewed by Youenn Fablet.

Follows 279811@main as another small incremental step towards fully
removing AppCache.

* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/page/DOMWindow.cpp:
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::applicationCache): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebKitLegacy/mac/WebKit.exp:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _cacheabilityDictionary]):
* Source/WebKitLegacy/mac/WebView/WebFramePrivate.h:

Canonical link: <a href="https://commits.webkit.org/279873@main">https://commits.webkit.org/279873@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d62719176f98fc9c6fa7dc3d6d684d7f45e55148

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54782 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34214 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7358 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58060 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5513 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57082 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41759 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5527 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44369 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3728 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56877 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32343 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47460 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25493 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29133 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4797 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3654 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/50951 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5014 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59650 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30028 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5160 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51792 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31170 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47542 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51197 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12040 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32179 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30958 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/page/DOMWindow.idl:
```diff
@@ -84,8 +84,6 @@
     readonly attribute Navigator navigator;
     [Replaceable, ImplementedAs=navigator] readonly attribute Navigator clientInformation;
-    // FIXME: This is specified to be [SecureContext]
-    [EnabledByQuirk=shouldEnableApplicationCache] readonly attribute DOMApplicationCache applicationCache;
```
Can we remove the quirk as well?

annevk replied:
No, it guards DOMApplicationCache too. And removing that I probably have to leave for @szewai.

annevk committed:
AppCache removal: remove applicationCache property
https://bugs.webkit.org/show_bug.cgi?id=275312

Reviewed by Youenn Fablet.

Follows 279811@main as another small incremental step towards fully
removing AppCache.

* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/page/DOMWindow.cpp:
* Source/WebCore/page/DOMWindow.h:
* Source/WebCore/page/DOMWindow.idl:
* Source/WebCore/page/LocalDOMWindow.cpp:
(WebCore::LocalDOMWindow::applicationCache): Deleted.
* Source/WebCore/page/LocalDOMWindow.h:
* Source/WebKitLegacy/mac/WebKit.exp:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame _cacheabilityDictionary]):
* Source/WebKitLegacy/mac/WebView/WebFramePrivate.h:

Canonical link: https://commits.webkit.org/279873@main
